javaslang/src/main/java/javaslang/collection/Array.java

```diff
@@ -1048,11 +1048,15 @@ public int length() {
 
     @Override
     public Array<T> reverse() {
-        final Object[] arr = new Object[delegate.length];
-        for (int i = 0; i < delegate.length; i++) {
-            arr[delegate.length - 1 - i] = delegate[i];
+        if (size() <= 1) {
+            return this;
+        } else {
+            final Object[] arr = new Object[delegate.length];
+            for (int i = 0; i < delegate.length; i++) {
+                arr[delegate.length - 1 - i] = delegate[i];
+            }
```

Comment:
If you think it's a bottleneck, we could create a `JMH` for it.
In my experience non-algorithmic complexity cannot be guessed without measurement! The JVM can eliminate simply eliminate it and reorder whatever it wishes, as long as the visible results are the same.